assembl/static2/js/app/components/common/flatList.jsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-undef */
+// @flow
+import React from 'react';
+import { I18n } from 'react-redux-i18n';
+
+import Loader from './loader';
+import { displayAlert } from '../../utils/utilityManager';
+import { APOLLO_NETWORK_STATUS } from '../../constants';
+
+type ItemNode = {
+  node: Object
+};
+
+type FlatListProps = {
+  items: {
+    pageInfo: {
+      endCursor: string,
+      hasNextPage: boolean
+    },
+    edges: Array<ItemNode>
+  },
+  ListItem: Function | ReactClass<*>,
+  className: string,
+  networkStatus: number,
+  onEndReachedThreshold: number,
+  itemData: Function,
+  fetchMore: Function,
+  refetch: Function,
+  extractItems: Function
+};
+
+class FlatList extends React.Component<*, FlatListProps, void> {
+  props: FlatListProps;
+
+  loading: boolean;
+
+  static defaultProps = {
+    onEndReachedThreshold: 0.8,
+    itemData: () => {}
+  };
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+  }
+
+  handleScroll = () => {
+    if (!this.loading) {
+      const { onEndReachedThreshold } = this.props;
+      // $FlowFixMe
```

Comment:
what are the flow problems here?

Reply:
Property cannot be accessed on possibly null value
null...